ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake

```diff
@@ -52,8 +63,13 @@ function(ament_target_dependencies target)
     target_include_directories(${target}
       PUBLIC ${ordered_include_dirs})
     ament_libraries_deduplicate(unique_libraries ${libraries})
-    target_link_libraries(${target}
-      ${unique_libraries})
+    if(AMENT_TARGET_DEPENDENCIES_PUBLIC)
+      target_link_libraries(${target}
+        PUBLIC ${unique_libraries})
+    else()
+      target_link_libraries(${target}
+        ${unique_libraries})
+    endif()
```

Comment:
Instead of duplicating the logic a variable (empty or containing `PUBLIC`) should be used and passed to `target_link_libraries`.